docs/en/qos.md

```diff
@@ -0,0 +1,252 @@
+
+To be noted, the share value configured on the table shall be reasonably large. As a contrary example, allocating a 1% quota ratio for a certain user may represent a granularity of GPU resource allocation too fine to serve an inference request, which could lead to starvation for this user.
+
+The below diagram shows a typical example of how this strategy works.
+
+![](https://github.com/InternLM/lmdeploy/assets/52888924/3e1d7135-6b11-4998-89a1-b72af6c962c3)
+
+#### Strategy 3: a combination strategy of 1 and 2
+
+We can call it a hybrid strategy. The way we hybrid these 2 strategies is fairly simple: we adopt strategy 1 in between user groups, and adopt strategy 2 within a user group. So users belonging to different groups with different priorities will only obey strategy 1 to determine their privilege in resource allocation. That is, when both strategies are applied, the first strategy will overpower the second. When it comes to a situation that no cross-group requests are pending for serving, the within-group strategy 2 comes to play.
+
+Below is a diagram showing it.
+
+![](https://github.com/InternLM/lmdeploy/assets/52888924/e335f976-ff15-48db-b1ff-abf1c3327d6e)
+
+To be noted, there could be other ways of hybrid strategies 1 & 2, this doc only introduced one method that works well in our scenario. Other hybrid methods shall consider that the prioritization and pro-rated sharing are obviously conflicting strategies so there isn't an easy way to mix them to work within a single dimension.
+
+That been said, there are ways to meet users' simple requirements to only make use of either one of the strategies. For example, a system requiring only a fair-sharing-like strategy can be configured with all users under one group. A good example is one model deployed for two different business purposes without dependency. A system having need of strictly chained prioritizations between different groups of users can be configured with at most 4 groups and each group having only one user. A good example is DAG inference workload.
+
+### A Sample QoS Configuration
+
+The configuration will be specified by the `--qos_config_path` flag, and will be loaded by program upon startup.
+
+```json
+{
+    "enable_user_qos": true,
+    "user_groups": [
+        "Platinum",
+        "Gold",
+        "Silver",
+        "Bronze"
+    ],
+    "user_group_map": {
+        "Platinum": [
+            {
+                "id": "user_id0",
+                "quota_pct": "100"
```

Comment:
seems `quota_pct` does not accept string

Reply:
Changed to `int`